File: ant-sdk/antd-dart/example/02_data.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:antd/antd.dart';

/// Demonstrates storing and retrieving public immutable data.
void main() async {
  final client = AntdClient();

  try {
    // Store data
    final data = Uint8List.fromList(utf8.encode('Hello, Autonomi!'));
    final result = await client.dataPutPublic(data);
    print('Stored at ${result.address}');
    print('  chunks: ${result.chunksStored}, mode: ${result.paymentModeUsed}');

    // Retrieve data
    final retrieved = await client.dataGetPublic(result.address);
    print('Retrieved: ${utf8.decode(retrieved)}');

    // Estimate cost
    final cost = await client.dataCost(data);
    print('Estimated cost: ${cost.cost} atto across ${cost.chunkCount} chunks');
  } on AntdError catch (e) {
    print('Error: $e');
  } finally {
    client.close();
  }
}

File: ant-sdk/antd-ruby/examples/02_data.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Example 02: Public data put/get with cost estimate

require_relative "../lib/antd"

client = Antd::Client.new

data = "Hello, Autonomi!"

# Estimate cost first
est = client.data_cost(data)
puts "Estimated cost: #{est.cost} atto (#{est.chunk_count} chunks)"

# Store data
result = client.data_put_public(data)
puts "Stored at #{result.address} (chunks: #{result.chunks_stored}, mode: #{result.payment_mode_used})"

# Retrieve data
retrieved = client.data_get_public(result.address)
puts "Retrieved: #{retrieved}"
